public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.ts

```diff
@@ -216,31 +223,30 @@ async function getParserCompletions(
   return completions;
 }
 
-async function getAfterSelectorCompletions(
+export async function getAfterSelectorCompletions(
   logQuery: string,
   afterPipe: boolean,
   hasSpace: boolean,
   dataProvider: CompletionDataProvider
 ): Promise<Completion[]> {
-  const { extractedLabelKeys, hasJSON, hasLogfmt } = await dataProvider.getParserAndLabelKeys(logQuery);
-
-  const prefix = `${hasSpace ? '' : ' '}${afterPipe ? '' : '| '}`;
-  const completions: Completion[] = await getParserCompletions(prefix, hasJSON, hasLogfmt, extractedLabelKeys);
+  let query = logQuery;
+  if (afterPipe) {
+    // If we have space after pipeline, we need to remove it as well
+    const trailingPipeline = hasSpace ? '| ' : '|';
+    query = trimEnd(logQuery, trailingPipeline);
```

Comment:
I would default to this, which works in both cases:
```suggestion
    query = trimEnd(logQuery, '| ');
```